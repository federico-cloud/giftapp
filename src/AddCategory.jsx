import React, { useState } from 'react';

export const AddCategory = ({setCategorias}) => {
  
    const [inputValue, setInputValue] = useState('Escriba aqui su categoria');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategorias( cats => [...cats, inputValue]);
            setInputValue('')
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}
