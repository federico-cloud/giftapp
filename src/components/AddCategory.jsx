import React, { useState } from 'react';

export const AddCategory = ({setCategorias}) => {
  
    const [inputValue, setInputValue] = useState('Escriba aqui su categoria');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategorias( cats => [inputValue,...cats]);
            setInputValue('')
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}
