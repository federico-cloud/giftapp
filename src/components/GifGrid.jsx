import React, { useEffect, useState } from 'react'

export const GifGrid = ({categoria}) => {

    const [imagenes, setImagenes] = useState([]);

    useEffect( () => {
        getGifs();
    }, [])

    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=goku&limit=20&api_key=FQNhGq9VGKl9uLcBgROf6bKArM2jKtTR';
        const resp = await fetch(url);

        const {data} = await resp.json();

        const gifs = data.map( img => {
            return{
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        setImagenes(gifs);
    } 
  
    return (
        <>
            <h3>{categoria}</h3>
            <ol>
                {
                    imagenes.map( img => (
                        <li key={img.id}> {img.title}</li>
                    ))
                }
            </ol>
        </>
    )
}
