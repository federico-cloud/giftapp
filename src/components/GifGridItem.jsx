import React from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ( { title ,url } ) => {

    return (
        <div className="card animate__animated animate__zoomIn">
            <p>{title}</p>
            <img src={url} alt={title}/>
        </div>
    )

}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}