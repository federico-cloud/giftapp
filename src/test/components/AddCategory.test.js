import React from 'react';
import {shallow} from 'enzyme';
import {AddCategory} from '../../components/AddCategory';

describe('Pruebas en el componenete <AddCategory />', () => {
    
    const setCategorias = jest.fn();
    let wrapper;

    beforeEach(() => {

        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategorias={setCategorias}/>);

    });


    test('Debe mostrarse correctamente', () => {    

        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';


        input.simulate('change', {
            target: {
                value: value
            }
        });
    })

    test('No debe postear la informacion con submit', () => {

        wrapper.find('form').simulate('submit',{ preventDefault(){} });

        expect( setCategorias ).toHaveBeenCalled();

    })

    test('Debe llamar el setCategorias y limpiar la caja de texto', () => {

        const value = 'Hola';

        wrapper.find('input').simulate('change', { target: {value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect( setCategorias ).toHaveBeenCalled();
        expect ( wrapper.find('input').prop('value')).toBe('');

    })

})