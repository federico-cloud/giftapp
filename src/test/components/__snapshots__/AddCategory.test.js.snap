// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componenete <AddCategory /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="Escriba aqui su categoria"
  />
</form>
`;
